src/stringify.spec.ts

```
@@ -44,3 +50,18 @@ describe("Stringify CSS spec compliance", () => {
 });
+
+describe("Stringify round-trips special characters in identifiers", () => {
+    const selectors = [
+        String.raw`.x\&y`,
+        String.raw`[data-foo\=bar]`,
+        String.raw`.a\{b\}c`,
+        String.raw`.w-1\/2`,
+        String.raw`#a\?b`,
+    ];
+    it.each(selectors)("%s", (selector) => {
+        expect(parse(stringify(parse(selector)))).toStrictEqual(
```

ESLint

[error] Call is nested too deeply. Maximum allowed is 3.

(unicorn/max-nested-calls)